packages/flame/lib/src/effects2/rotate_effect.dart

```diff
@@ -0,0 +1,50 @@
+import 'effect_controller.dart';
+import 'transform2d_effect.dart';
+
+/// Rotate a component around its anchor.
+///
+/// Two constructors are provided:
+///   - [RotateEffect.by] will rotate the target by the specified `angle`
+///     relative to its orientation at the onset of the effect. For example,
+///     rotating by `angle = tau/4` will turn the component 90° clockwise
+///     relative to its initial direction;
+///   - [RotateEffect.to] will rotate the target to the fixed orientation
+///     specified by the `angle`. For example, rotating to `angle = tau/4` will
+///     turn the component to look East regardless of its initial bearing.
+///
+/// This effect applies incremental changes to the component's angle, and
+/// requires that any other effect or update logic applied to the same component
+/// also used incremental updates.
+class RotateEffect extends Transform2DEffect {
+  RotateEffect.by(double angle, EffectController controller)
+      : _angle = angle,
+        super(controller);
+
+  factory RotateEffect.to(double angle, EffectController controller) {
```

Comment:
Commenting here instead since it has more with this line to do than the ongoibg discussing further down.

Will it not be clearer to have an argument here which specifies whether it goes the shortest path or not? `to` vs. `towards` doesn't really make it clear which one that goes the shortest way I think. Or `towards` is pretty clear, but not `to`, possibly a better name could also be used?

Reply:
So, here was my thinking: 
I started with a `MoveEffect`, which has two flavors: `MoveEffect.to` which moves to a specific point on screen, and `MoveEffect.by` which shifts the current position by the specified offset. The first name is nice and clear, the second sounds a bit weird, but I couldn't think of anything better. Then came the `RotateEffect`. By analogy, I made `RotateEffect.by` (which is now nice and clear), and `RotateEffect.to` (which is ok).

I imagine we'd add `ScaleEffect.to()` / `ScaleEffect.by()` later on too. And it would be nice to have all these names consistent with each other.

Now, `RotateEffect.towards()` is different from the rest, but that's because rotation has this weird property that 0º==360º.

Having said that, I agree that the distinction between `towards` and `to` may not be immediately clear to a casual reader. But the mnemonics is clear here: `towards`= in general direction of, whereas `to`= go there. The use cases are also quite different: you would use `.towards` in a game like space-shooter, where you need to orient your character or projectiles in the direction of an enemy; while `.to` has more niche purposes, such as rotary phones, various knobs on a control panel, roulette, spin-wheel, etc.

Reply:
Alright, I can't come up with any better name either, and i know that you'll document it well so it should be fine.